File: python/examples/basic_chat.py
"""Blocking chat against a coordinator (or a solo node — same address, same client).

INFERHUB_BASE=http://localhost:5080/ INFERHUB_API_KEY=... python examples/basic_chat.py
"""

import os

from inferhub_client import ChatMessage, ChatRequest, InferHubClient

base_url = os.environ.get("INFERHUB_BASE", "http://localhost:5080/")
api_key = os.environ.get("INFERHUB_API_KEY")

with InferHubClient(base_url, api_key) as client:
    answer = client.chat(
        ChatRequest(
            model="llama3",
            messages=[ChatMessage(role="user", content="Say hi in one word.")],
        )
    )
    print(answer.message.content)

File: js/examples/basic-chat.ts
/**
 * Blocking chat against a coordinator (or a solo node — same address, same client).
 *
 * INFERHUB_BASE=http://localhost:5080/ INFERHUB_API_KEY=... npx tsx examples/basic-chat.ts
 * (or: npm run build && node --experimental-strip-types examples/basic-chat.ts on Node ≥ 22)
 */

import { InferHubClient } from "../src/index.js";

const baseUrl = process.env.INFERHUB_BASE ?? "http://localhost:5080/";
const apiKey = process.env.INFERHUB_API_KEY;

const client = new InferHubClient({ baseUrl, apiKey });

const answer = await client.chat({
  model: "llama3",
  messages: [{ role: "user", content: "Say hi in one word." }],
});

console.log(answer.message?.content);
